### PR TITLE
Redact tokens/passwords from URLs in ConsDbClient HTTPError reports

### PR DESCRIPTION
**Pull request by Vebop**
## Added `clean_url()` to `requests.Session` hooks for responses. 
Added test to check initialization of ConsDbClient and to test that the url is sanitized according to our terms.

### CODE REVIEW

**Comment by bbrondel** [comment]
I'd think you would want to redact the entire password, including the first two letters, and for that matter avoid revealing anything about the length of the password.

**Comment by mfisherlevine** [comment]
Also, I'm not usually a stickler for asking people to squash commits, but given the middle commits include adding and removing a whole file which should never have been committed, and that this whole ticket feels fine as a single commit, I'd probably just squash the whole thing down to one or two commits.

File: python/lsst/summit/utils/consdbClient.py [hunk 1]
```diff
@@ -77,6 +77,24 @@ def _check_status(r: requests.Response) -> None:
...
+
+    Set new url in response and return response
+
+    Parameters
+    ----------
+    resp : `requests.Response`
+        The response that could contain a URL with tokens
+    """
+    url = urlparse(resp.url)
+    short_user = url.username[:2] if url.username is not None else "**"
```
Comment (ktlim) [inline python/lsst/summit/utils/consdbClient.py:91]: I think it's a bit more useful to have
```
short_user = f"{url.username[:2]}***" if url.username is not None else ""
short_pass = f":{url.password[:2]}***" if url.password is not None else ""
netloc = f"{short_user}{short_pass}@{url.hostname}"
```
That way, empty usernames and passwords are properly reflected and the password component is only included if it is actually present.

File: tests/test_consdbClient.py [hunk 2]
```diff
@@ -28,6 +28,9 @@
 
 @pytest.fixture
 def client():
+    """Initilaize client with a fake url
```
Comment (ktlim) [inline tests/test_consdbClient.py:31]:
```suggestion
    """Initialize client with a fake url
```

File: tests/test_consdbClient.py [hunk 3]
```diff
@@ -161,6 +164,38 @@ def test_schema(client):
...
+        )
+
+    url = error.value.args[0].split()[-1]
+    sanitized = (
+        "https://us***:v9***@usdf-fake.slackers.stanford.edu/consdb/flex/bad_instrument/exposure/addkey"
+    )
+
+    assert "v987wefVMPz" not in url
+    assert url == sanitized
+
```
Comment (ktlim) [inline tests/test_consdbClient.py:192]: If you really want coverage, you could also test the username/no password case, a password/no username case (although I don't know if that is actually legal), and a single-character username case (if someone uses 1 or 2 character passwords, that's their own problem...).

**Review by ktlim — APPROVED** [review]
General approach looks great.

**Comment by Vebop** [comment]
Updated the `short_user` and `short_pass` to be empty if none are provided; 
Checked multiple test cases in a parametrized unit test; 
Reorganized commits -- but don't we use a squash commit :)

**Pull request merged**